src/Components/Components/src/CascadingValue.cs

```diff
@@ -76,10 +76,7 @@ public Task SetParametersAsync(ParameterView parameters)
             else if (parameter.Name.Equals(nameof(Name), StringComparison.OrdinalIgnoreCase))
             {
                 Name = (string)parameter.Value;
-                if (string.IsNullOrEmpty(Name))
-                {
-                    throw new ArgumentException($"The parameter '{nameof(Name)}' for component '{nameof(CascadingValue<TValue>)}' does not allow null or empty values.");
```

Comment:
This was clearer with the old exception message. `parameter.Name` is not null. `parameter.Name == "Name"` and `string.IsNullOrEmpty(parameter.Value)`.

Let's only replace messages where the new message is at least as good or better.

Reply:
Fair enough, I reverted this change.